<launch>

  <arg name="world_frame_id" default="world_enu"/>
  <arg name="odom_topic" default="/airsim_node/drone_1/odom_local_enu"/>
  <arg name="cmd_topic" default="/quadrotor/position_cmd"/>
  <arg name="waypoints_topic" default="/quadrotor/waypoints"/>
  <arg name="mav_name" />

  <!-- random map -->
  <arg name="global_map_topic" default="/random_forest/global_cloud"/>

  <arg name="map_size_x" default="40"/>
  <arg name="map_size_y" default="20"/>
  <arg name="map_size_z" default="10"/>

  <!-- main algorithm params -->
  <include file="$(find plan_manage)/launch/kino_algorithm.xml">
    <arg name="world_frame_id" value="$(arg world_frame_id)"/>

    <arg name="map_size_x_"    value="$(arg map_size_x)"/>
    <arg name="map_size_y_"    value="$(arg map_size_y)"/>
    <arg name="map_size_z_"    value="$(arg map_size_z)"/>

    <arg name="odom_topic"      value="$(arg odom_topic)"/>
    <arg name="waypoints_topic" value="$(arg waypoints_topic)"/>

    <arg name="tracker_topic" value="/$(arg mav_name)/trackers_manager/bspline_tracker/BsplineTracker/goal"/>
    <arg name="srv_name"  value="/$(arg mav_name)/mav_services/bspline_tracker"/>

    <!-- camera pose: transform of camera frame in the world frame -->
    <!-- depth topic: depth image, 640x480 by default -->
    <!-- don't set cloud_topic if you already set these ones! -->
    <arg name="camera_pose_topic" value="/quad_pail/converted_camera_pose"/>
    <arg name="depth_topic" value="/airsim_node/drone_1/depth_vis/DepthVis"/>

    <!-- topic of point cloud measurement, such as from LIDAR  -->
    <!-- don't set camera pose and depth, if you already set this one! -->
    <arg name="cloud_topic" value="/airsim_node/drone_1/DepthPerspective/registered/points"/>

    <!-- intrinsic params of the depth camera -->
    <arg name="cx" value="320.0"/>
    <arg name="cy" value="240.0"/>
    <arg name="fx" value="320.0"/>
    <arg name="fy" value="320.0"/>

    <!-- maximum velocity and acceleration the drone will reach -->
    <arg name="max_vel" value="3.0" />
    <arg name="max_acc" value="2.0" />

    <!-- 1: use 2D Nav Goal to select goal  -->
    <!-- 2: use global waypoints below  -->
    <arg name="flight_type" value="1" />
    
    <!-- global waypoints -->
    <!-- If flight_type is set to 2, the drone will travel these waypoints one by one -->
    <arg name="point_num" value="2" />

    <arg name="point0_x" value="19.0" />
    <arg name="point0_y" value="0.0" />
    <arg name="point0_z" value="1.0" />

    <!-- set more waypoints if you need -->
    <arg name="point1_x" value="-19.0" />
    <arg name="point1_y" value="0.0" />
    <arg name="point1_z" value="1.0" />

    <arg name="point2_x" value="0.0" />
    <arg name="point2_y" value="19.0" />
    <arg name="point2_z" value="1.0" />
    
  </include>


  <node pkg="local_sensing_node" type="pcl_render_node" name="pcl_render_node" output="screen">
    <rosparam command="load" file="$(find local_sensing_node)/params/camera.yaml" />
    <param name="sensing_horizon"  value="5.0" />
    <param name="sensing_rate"     value="30.0"/>
    <param name="estimation_rate"  value="30.0"/>

    <param name="map/x_size"     value="$(arg map_size_x)"/>
    <param name="map/y_size"     value="$(arg map_size_y)"/>
    <param name="map/z_size"     value="$(arg map_size_z)"/>

    <remap from="~global_map" to="$(arg global_map_topic)"/>
    <remap from="~odometry"   to="$(arg odom_topic)"/>
  </node>

</launch>
